### PR TITLE
Add support in resample for gwcs.WCS.bounding_box

### PR DESCRIPTION
**Pull request by jdavies-st**
Please don't merge yet, more changes coming.

@nden could you look at the updates to `assign_wcs.util.wcs_from_footprints()`?

### CODE REVIEW

File: jwst/assign_wcs/util.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def wcs_from_footprints(dmodels, refmodel=None, transform=None, bounding_box=Non
         rotation = astmodels.AffineTransformation2D(np.array(wcsinfo['PC']))
         transform.append(rotation)
         if sky_axes:
-            scale = wcsinfo['CDELT'][sky_axes].mean()
+            cdelt1, cdelt2 = wcsinfo['CDELT']
```
Comment (nden) [inline jwst/assign_wcs/util.py]: If you intend to use the CDELT corresponding to the celestial axes you should use

`cdelt1, cdelt2 = wcsinfo['CDELT'][sky_axes]`

This ensures that you are always ppicking the celestial axes regardless of the order of axes.

Comment (nden) [inline jwst/assign_wcs/util.py]: I mean in some cases wcsinfo['CDELT'] may have 3 values.

File: jwst/resample/resample_utils.py [hunk 2]
```diff
@@ -135,7 +134,7 @@ def calc_gwcs_pixmap(in_wcs, out_wcs):
     """ Return a pixel grid map from input frame to output frame.
     """
     # TODO: Is the following 1-indexed or 0-indexed?  Check.
-    grid = wcstools.grid_from_domain(in_wcs.domain)
+    grid = wcstools.grid_from_bounding_box(in_wcs.bounding_box, step=(1, 1))
```
Comment (nden) [inline jwst/resample/resample_utils.py]: Is it also necessary to use `center=True` to get the grid at the center of the pixels?

**Comment by nden** [comment]
@jdavies-st lgtm

File: jwst/assign_wcs/util.py (hunk 1 above)
Comment (jdavies-st) [inline jwst/assign_wcs/util.py]: Right.  Thanks for the heads-up.  Will make this change.

**Comment by jdavies-st** [comment]
Minor changes on this last commit.  Disabling CI checks and merging.

**Pull request merged**